dev-libs/crypto++/crypto++-5.6.5.recipe

```diff
@@ -48,24 +53,39 @@ BUILD_PREREQUIRES="
 	cmd:gcc$secondaryArchSuffix
 	cmd:ld
 	cmd:make
-	cmd:pkg_config$secondaryArchSuffix
 	cmd:nasm
+	cmd:pkg_config$secondaryArchSuffix
 	"
 
+defineDebugInfoPackage crypto++$secondaryArchSuffix \
+	$libDir/libcryptopp.so.$libVersion
+
 BUILD()
 {
-	make shared dynamic static $jobArgs
+	make HAS_SOLIB_VERSION=1 dynamic static $jobArgs
 }
 
 INSTALL()
 {
-	make install PREFIX=$prefix INCLUDEDIR=$includeDir
-
-	mv $libDir/libcryptopp.so $libDir/libcryptopp.so.$portVersion
-	ln -s $libDir/libcryptopp.so.$portVersion $libDir/libcryptopp.so
+	make \
+		HAS_SOLIB_VERSION=1 LDCONF=true \
+		BINDIR=$binDir LIBDIR=$libDir \
+		DATADIR=$dataDir INCLUDEDIR=$includeDir \
+		install
 
+	# Yes, they created the link. But they forgot to install it...
+	ln -s libcryptopp.so.$libVersion \
+		$libDir/libcryptopp.so.${libVersionCompat##* }
 	prepareInstalledDevelLib libcryptopp
 
+	install -d -m 755 $developLibDir/pkgconfig
+	cat > $developLibDir/pkgconfig/libcrypto++.pc << EOF
+Name: libcrypto++
+Description: Class library of cryptographic schemes
+Version: $portVersion
+Libs: -lcryptopp
```

Comment:
Thanks @alaviss for your recent updates. I have a last suggestion...

I understood that the following suggestion is not needed, but I'd like to suggest it for 2 reasons (explained below):
```diff
 	cat > $developLibDir/pkgconfig/libcrypto++.pc << EOF
+prefix=$prefix
+exec_prefix=\${prefix}
+libdir=\${prefix}/$relativeDevelopLibDir
+includedir=\${prefix}/$relativeIncludeDir
 Name: libcrypto++
 Description: Class library of cryptographic schemes
 Version: $portVersion
-Libs: -lcryptopp
+Libs: -L\${libdir} -lcryptopp
 EOF
```
* This would make the pc file (for this lib) consistent with the pc files that are generated for all the other packages.
* Paranoids like me would feel safer ;-)

The only downside would be that if/when pkg-config ever changes in a way that it stops adding all these lines, then we would have to revert this suggestion to keep the consistency.
What do you think?
Isn't it nice? (I've even "converted" it to avoid the additional-files directory ;-)

Reply:
Oops, I forgot to tell that everything is OK, so you can add a ` (#1900)` to your commit summaries like we did for signify.
I hope you will accept my last suggestion, but, of course, I can merge this PR without it if you prefer and/or if you convince me that it is totally unneeded.

Reply:
IMO, consistency is unnecessary for a file that most humans will never read. They appear "consistent" as they are usually generated with autotools using templates like [this][0]

Those additional lines, which are freeform variable, have no special meaning in a pkg-config file. They are often used to [reduce clutter][1]
> Before describing the more useful fields, it will be helpful to demonstrate variable definitions. The most common usage is to define the installation paths so that they don't clutter the metadata fields. Since the variables are expanded recursively, this is very helpful when used in conjunction with autoconf derived paths.

Furthermore, since our library was put in Haiku's compiler search path (`/boot/system/develop/lib`), the `-L` directive (which tells the compiler where to find said library) becomes redundant in this case, which make those variables become more or less clutter themselves
```
$ gcc -print-search-dirs
**omitted**
libraries: **lots of stuff omitted**:/boot/system/develop/lib/
```

As for you being paranoid, be sure that all ports I made and/or modified are tested on both x86_gcc2, x86 (secondary) and x86_64 so they should always behave correctly

[0]: https://tecnocode.co.uk/2014/12/09/a-checklist-for-writing-pkg-config-files/
[1]: https://people.freedesktop.org/~dbn/pkg-config-guide.html#writing

Reply:
OK, thanks for the pointers and explanation :)